src/types/YArray.js

```diff
@@ -201,6 +203,22 @@ export class YArray extends AbstractType {
     return typeListGet(this, index)
   }
 
+  /**
+   * Returns the weak link to i-th element from a YArray.
+   *
+   * @param {number} index The index of the element to return from the YArray
+   * @return {YWeakLink<T>}
+   */
+  link(index) {
+    if (this.doc !== null) {
```

Comment:
I propose to limit WeakLinks only to refer to types. This would simplify the implementation.

Mainly, I want to avoid having a list of WeakLinks references on each type. Alternatively, WeakLink could only store the ID and look up the item every time the value is requested (if the item is deleted, then we return undefined).